### PR TITLE
Add ExtractorResult --> SeriesStim converter

### PR DESCRIPTION
**Pull request by tyarkoni**
This PR adds an `ExtractorResultToSeriesConverter` class that converts an `ExtractorResult` instance into a list of `SeriesStim` instances. The `SeriesStim` is also new, and is just a wrapper around a pandas `Series`. This allows us to build extractors that operate on the outputs of other extractors (e.g., to compute summary statistics over multiple columns). It's a bit inelegant in that it required a bit (but only a bit) of *sui generis* handling of `ExtractorResult` inputs (since we can otherwise assume `Transformer` inputs are always `Stim` subclasses), but I think it's a small price to pay for the added flexibility.

@rbroc, in principle, the existing automatic conversion detection should kick in without having to do anything further, so any `Extractor` that specifies a `SeriesStim` input type *should* be able to take an `ExtractorResult` as input to `transform()`. But I haven't tested this yet, so let me know if you run into problems.

Note that I ended up splitting the `ExtractorResult` DFs into a separate `SeriesStim` for each row after all (much like your `VectorStim`), as there would otherwise have been some thorny issues to deal with. Also, there isn't really any way I can see to distinguish between "good" and "bad" features on the `Stim` side of things; I think this will need to be built into the `MetricExtractor` hierarchy, unless we want to rework the `ExtractorResult` class to have some concept of metadata (that's aspirational, but probably not happening any time soon). For example, for the BertLM extractor, if you leave metadata on, you're going to get not only one column per prediction, but also a "sequence" column. The extractor will then need to have some way of knowing which features to ignore—perhaps by taking this as an initialization argument. I also don't currently do any checking for data type, which is likely to be problematic. But again I would prefer to do this on the extractor side rather than the conversion/stim side, as there could be principled reasons for wanting to retain, e.g., string columns in the `SeriesStim`, even if most extractors don't know what to do with that.

### CODE REVIEW

**Comment by rbroc** [comment]
@tyarkoni looks good and works with updated MetricExtractor (still in separate PR), which is now a misc type extractor.
Will look into it thoroughly again tomorrow, but a couple of very minor comments/questions in the meantime:
- Re: `ExtractorResultToSeriesConverter`. It always returns a list of `SeriesStim`, even for 1-row `ExtractorResults`. Could/should de-listify the `SeriesStim` in the latter case? It'd allow to use `to_df` on the metric extractor result, which may be more intuitive? 
- Re:`SeriesStim`, very minor thing, but should we allow to read a labels/index column when filename is specified, to be used as index for the series and gain more flexibility when subsetting to relevant columns in `Extractor` init call? 
- I've added to `MetricExtractor `option to subset `SeriesStim.data` by a list passed at init stage (must be subset of series indices or positional indices). However, the issue of not being able to disentangle features and metadata in ExtractorResult has popped up a few times lately, so maybe worth thinking about it soon.


- Are dropping the idea of a DataFrameStim for now? In the longer run, it may be nice to operate over columns too, e.g. if we're open to having extractors that compute autoregressive metrics. Another idea could be having `DataFrameStim` as a collection of `SeriesStim` (same logic as `ComplexTextStim` to `TextStim`).

**Comment by adelavega** [comment]
@rbroc About the Series labels, the way I understood it is that its somehting you should be able to control via kwargs to `pd.read_csv`

**Comment by rbroc** [comment]
> @rbroc About the Series labels, the way I understood it is that its somehting you should be able to control via kwargs to `pd.read_csv`

oh, right, sorry. Had completely overlooked that there were kwargs! 🙏

**Comment by tyarkoni** [comment]
> Re: ExtractorResultToSeriesConverter. It always returns a list of SeriesStim, even for 1-row ExtractorResults. Could/should de-listify the SeriesStim in the latter case? It'd allow to use to_df on the metric extractor result, which may be more intuitive than merge_results?

Tough call. You get unintuitive/annoying behavior either way. If you return a single `SeriesStim`, you force the end user's code to type-check or add a conditional any time they don't know how many input rows there were. If you always return a list, they have to index into it explicitly any time they know there's only one input row. I'm not sure which is the lesser of two evils, and I don't remember how we've handled this elsewhere (internal consistency could be the tie-breaker).

> I've added to MetricExtractor option to subset SeriesStim.data by a list passed at init stage (must be subset of series indices or positional indices). However, the issue of not being able to disentangle features and metadata in ExtractorResult has popped up a few times lately, so maybe worth thinking about it soon.

Agreed.

> Are dropping the idea of a DataFrameStim for now? In the longer run, it may be nice to operate over columns too, e.g. if we're open to having extractors that compute autoregressive metrics. Another idea could be having DataFrameStim as a collection of SeriesStim (same logic as ComplexTextStim to TextStim).

There could definitely be use cases for a `DataFrameStim` in the future, but I'd rather not add it preemptively—let's wait until we actually need it.

Thanks!

File: pliers/converters/misc.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class ExtractorResultToSeriesConverter(Converter):
+    """Converts an ExtractorResult instance to a list of SeriesStims."""
+
+    _input_type = ExtractorResult
+    _output_type = SeriesStim
+
+    def _convert(self, result):
+        df = result.to_df(timing=False, metadata=False, object_id=False)
```
[the remark this answers is not in the document]
Comment (tyarkoni) [reply pliers/converters/misc.py:15]: I'll leave it as-is for now; we can revisit later if needed.

**Pull request merged**